tests/unit_tests/lcls_tools/common/measurements/test_emittance_measurement.py

```diff
@@ -103,34 +89,127 @@ def mock_function(scan_settings, function):
+                    mock_beamsize_measurements += [result]
+
+                # External list to return beam sizes
+                external_list = iter(mock_beamsize_measurements)
+
+                # Mock beamsize_measurement
+                mock_beamsize_measurement = MagicMock(spec=ScreenBeamProfileMeasurement)
+                mock_beamsize_measurement.device = MagicMock(spec=Screen)
+                mock_beamsize_measurement.device.resolution = 1.0
+                mock_beamsize_measurement.measure = MagicMock(
+                    side_effect=lambda _: next(external_list)
+                )
+
+                # Mock magnet
+                mock_magnet = MagicMock(spec=Magnet)
+                mock_magnet.metadata = MagnetMetadata(
+                    area="test", beam_path=["test"], sum_l_meters=None, l_eff=0.1
+                )
+
+                def mock_function(scan_settings, function):
+                    for ele in scan_settings:
+                        function()
+
+                mock_magnet.scan = mock_function
+
+                # Instantiate the QuadScanEmittance object
+                quad_scan = QuadScanEmittance(
+                    energy=1e9 * 299.792458 / 1e3,
+                    scan_values=k,
+                    magnet=mock_magnet,
+                    beamsize_measurement=mock_beamsize_measurement,
+                    n_measurement_shots=1,
+                    rmat=rmat,
+                    design_twiss=design_twiss_ele,
+                    wait_time=1e-3,
+                )
+
+                # Call the measure method
+                result = quad_scan.measure()
+
+                # Assertions
+                assert isinstance(result, EmittanceMeasurementResult)
+                assert hasattr(result, "rms_x")
+                assert hasattr(result, "rms_y")
+                assert len(result.rms_x) == len(quad_scan.scan_values)
+                assert len(result.rms_y) == len(quad_scan.scan_values)
+
+                # check resulting calculations against cheetah simulation ground truth
+                assert np.allclose(
+                    result.emittance,
+                    np.array([1.0e-2, 1.0e-1]).reshape(2, 1),
+                )
+                assert np.allclose(
+                    result.beam_matrix,
+                    np.array([[5.0e-2, -5.0e-2, 5.2e-2], [0.3, -0.3, 0.33333328]]),
+                )
+
+                if design_twiss_ele is not None:
+                    assert np.allclose(result.BMAG[:, 4], 1.0)
+
+                    # test get_best_bmag method
+                    for mode in ["x", "y", "geometric_mean"]:
+                        best_bmag = result.get_best_bmag(mode)
+                        assert np.allclose(best_bmag[0], result.quadrupole_strengths[4])
+                        assert np.allclose(best_bmag[1], 1.0)
+                else:
+                    with self.assertRaises(ValueError):
```

Comment:
ValueError is too coarse of a test. It might be raised for all kinds of reasons. Rather, use `assertRaiseRegex` which allows you test the reason for the exception. Again, I probably wouldn't test this specific case. It's unlikely to be broken.